### ISSUE TITLE
IMAP login can be restricted to a specific domain name.

### ISSUE THREAD

**Issue by kosli** [nextcloud/apps#1]
Not sure where i should re-create my issue from owncloud?
 --> https://github.com/owncloud/apps/pull/2093

**Comment by t2d** [comment]
I would rather like to see this implemented: https://github.com/owncloud/apps/pull/2166

**Comment by t2d** [comment]
In my instance user can login the nextcloud through imap auth with `username` or `username@domain`. Either way, nextcloud adds the users as `username` internally.
Sadly, this means, that later Cal/CardDAV auth also only works with `username`.
I believe, we should also add the optional domain to `apps/dav/lib/Connector/PublicAuth.php`.

**Comment by nickvergessen** [comment]
Yeah @kosli please make a PR on our end here, if that helps to solve an issue.

**Comment by kosli** [comment]
@nickvergessen: done :-)

**Comment by xshadow** [comment]
+1
